Tasks/SonarQubePostTest/SummaryReport/QualityGateSection.ps1

```diff
@@ -0,0 +1,222 @@
+
+function GetQualityGateSectionContent
+{
+    Write-Verbose "Formatting the quality gate report section"
+    
+    WaitForAnalysisToFinish
+    
+    $qualityGateResponse = FetchQualityGateDetails
+    $qualityGateStatus = $qualityGateResponse.projectStatus.status
+    $section1 = FormatQualityGateStatusSection $qualityGateStatus
+    
+    $warningsAndErrors = GetQualityGateWarningsAndErrors $qualityGateResponse    
+    $section2 = FormatQualityGateMessagesSection $warningsAndErrors
+    
+    $content = FormatQualityGateSection $section1 $section2
+    return $content
+}
+
+function FormatQualityGateSection
+{
+    param ($section1, $section2)
+    
+    return "$section1 $section2"
+}
+
+function FormatQualityGateMessagesSection
+{
```

Comment:
It's not easy to see what is supposed to be in each section: a short comment giving an example of the what the section is supposed to contain would help e.g.

Example output:
Duplicated blocks 1 < 10
Technical debt on new code  99 < 200